### DISCUSSION TITLE
Multiple layer picked from the same layer.

### DISCUSSION THREAD

**Discussion by mrsgrooovy** [nftchef/art-engine#136] [Help + Q&A]
I have created Hairs as a layer and generated the image but instead of picking one image from this layer it picks 2 or 3 images from the same Layer folder. Some of the names are same with different weighage with # delimiter.

Why would Hair have 2 jsons from where ?

    {
      "trait_type": "C:\\Users\\HP\\Downloads\\Codes\\art-engine-nested-folder-structure\\art-engine-nested-folder-structure\\layers",
      "value": "Hair"
    },
    {
      "trait_type": "Hair",
      "value": "Dark Brown Bun"
    },

Has anyone faced this ?

**Comment by BakeSloth** [comment]
you have to make sure that your folders and file names are weighted.

Trait_Subfolder#50
Trait_Filename.png#50

where 50 mean 50%

weights are required, or you will have the issue you are seeing

**Comment by cryptofannatic** [comment]
Hey @BakeSloth, I'm facing the same issue. Just to be clear, you mean we need to add weights to the folders just against the folder name right? or do we need to do anything in the layer configurations too inside the code? Because i just added weights to the sub folders like shown in the screenshot, and it's giving me this error after generating the first image
![Screenshot 2022-07-10 at 2 58 13 AM](https://user-images.githubusercontent.com/109002076/178123
![Screenshot 2022-07-10 at 2 58 59 AM](https://user-images.githubusercontent.com/109002076/178123280-261534b0-4f06-4eb7-a050-831ab2d53cce.png)
254-28bcaae0-5cdd-4a1d-b8bb-3fbb68bb322d.png)

Edit: I actually managed to figure out why i was getting the error, it was because of this function -  resetNameIndex:
